assets/app/view/game/stock_market.rb

```diff
@@ -338,9 +338,11 @@ def render
           type_text = @game.class::MARKET_TEXT
           colors = @game.class::STOCKMARKET_COLORS
 
-          types_in_market = @game.stock_market.market.flatten.compact.flat_map(&:types)
-          .uniq.to_h { |p| [p, colors[p]] }
+          types_in_market_map = @game.stock_market.market.flatten.compact.flat_map(&:types).uniq
+          types_in_market_map.sort! if @game.stock_market.sort_market_legend?
+          types_in_market_map.reverse! if @game.stock_market.reverse_market_legend?
```

Comment:
Seems pretty complicated and confusing to use. I recommend using the order as defined in @game::class.MARKET_TEXT. It's easy for a game to change that order if they want to.

Reply:
done, however this will impact games where the MARKET_TEXT is different than the order of prices as appearing the market (46 for example), the legend will be in the wrong order, but this is easily fixed.

Reply:
should i add a flag to use this? or just apply everywhere?